### DISCUSSION TITLE
New line

### DISCUSSION THREAD

**Discussion by bsastregx** [markedjs/marked#2631] [Q&A]
Hi, thanks so much for such an awesome package.

I am using [Strapi](https://strapi.io/) for my backend, which is a headless cms.  It is an awesome tool. 
With Strapi you add information/data on the backend, and it exposes an API with all the data, which you can consume on the front end.

Strapi has a rich text editor, which is exposed as a markdown string on the API.

I need to transform that markdown string into HTML.
I am succesfully doing that, but, the new lines are not being interpreted. 

Strapi creates a \n for each new line, and aparently marked is not interpreting it. It is displaying the \n as a string.

Any help will be much appreciated.

![image](https://user-images.githubusercontent.com/54677414/198896503-cd60e959-d544-4c5f-99af-2430a9dc48dc.png)

**Comment by UziTech** [comment]
Can you provide some code? And provide the actual string sent to marked?

**Comment by UziTech** [comment]
Also if you think this is a bug in marked it is better to create a new issue instead of a discussion.

**Reply by bsastregx** [reply]
Hi @UziTech I don't know yet is this is a bug or is my mistake.
After I am positive this is a bug I will create a new issue.

So, this is the string I am passing to the component via a "description" property: 

```description="With the new version, **GeneXus 18**, it is possible to create new and colossal digital ecosystems, integrate with these in an outsourced way to speed up your solution and build Scalable Mission Critical Applications in a safe, fast and automatic way.  \n With the new version, _GeneXus 18, [it is possible to create new and](www.google.com) colossal _digital ecosystems, integrate with these in an outsourced way to speed up your solution and build Scalable Mission Critical Applications in a safe, fast and automatic way."```

This property expects a string. I know that marked is working fine, because the output is ok, the only thing missing is the new line.

This is the code:

`{this.description ? <div innerHTML={marked.parse(this.description)}></div> : <slot></slot>}`

And this is the front end output:

`<p>With the new version, <strong>GeneXus 18</strong>, it is possible to create new and colossal digital ecosystems, integrate with these in an outsourced way to speed up your solution and build Scalable Mission Critical Applications in a safe, fast and automatic way.  \n With the new version, _GeneXus 18, <a href="www.google.com">it is possible to create new and</a> colossal _digital ecosystems, integrate with these in an outsourced way to speed up your solution and build Scalable Mission Critical Applications in a safe, fast and automatic way.</p>`

Thanks so much for such a quick response.

**Reply by UziTech** [reply]
Try to console.log `this.description` and `marked.parse(this.description)` to see if you can see where the `\n` is changed to `\\n`.

**Reply by bsastregx** [reply]
For the purpose of this example I have shorten the description to be "My name is \n Bruno".

This is the result of the console.log of `this.description` : My name is \n Bruno
This is the result of the console.log of `marked.parse(this.description)` : `<p>My name is \n Bruno</p>`

**Reply by UziTech** [reply]
If `console.log(this.description)` shows `My name is \n Bruno` in the console that means `\n` is the string `"\\n"`. If it were actually a new line character it would show as:

```
My name is 
 Bruno
```

in the console.

You can test it by doing `console.log(this.description.includes('\n'))`. Which will probably be false.

So it looks like the string marked is getting doesn't have a newline character in it which means it is getting changed before being sent to marked.

**Reply by bsastregx** [reply]
@UziTech Thanks very much for pointing that out. What you are saying makes sense. However, I just tried parsing a new line with marked, and it does not appears to be working: 

`<div innerHTML={marked.parse('My name is \n Bruno')}></div>`

![image](https://user-images.githubusercontent.com/54677414/198902813-6b689989-7814-459e-9a80-dd5fe8ab0c9d.png)

Please, correct me if I am wrong.

**Reply by UziTech** [reply]
Ya it looks like it isn't marked that is changing `\n` to `\\n` it is something before the string gets to marked. I would create an issue at Strapi to see if they can find where it is getting changed.

**Reply by UziTech** [reply]
By the way the reason you aren't seeing `Bruno` on a new line after it goes to marked is because that is what markdown is supposed to do. A new paragraph needs to be preceded by two `\n` characters. Alternatively you can set the `breaks` option and `gfm` option to true in marked.

**Reply by bsastregx** [reply]
Thanks for pointing that out. Strapi is doing a good work. It puts \n\n when a lineabreak is present. 
It was me who was trying out with just one /n, because I didn't know how markdown works. 
I think the problem is in StencilJS:

If i pass the string as a property to the component:

`description="My name is \n\n Bruno"`
`<div innerHTML={marked.parse(this.description)}></div>`

![image](https://user-images.githubusercontent.com/54677414/198904243-a1d6988f-7986-4bea-a8bf-42ccf20caaac.png)

If I pass the string directly to marked:
`<div innerHTML={marked.parse('My name is \n\n Bruno')}></div>`

![image](https://user-images.githubusercontent.com/54677414/198904284-3e97c607-4335-46b8-9647-41e95c6a0ce9.png)

Thanks so much for all the help during the day. 😘
